perpVects/QFPHelpers.cpp

```diff
@@ -37,19 +37,51 @@ std::ostream& operator<<(std::ostream& os, const unsigned __int128 &i){
   return os;
 }
 
-std::string
-getSortName(sort_t val){
-  switch(val){
-  case lt:
-    return "lt";
-  case gt:
-    return "gt";
-  case bi:
-    return "bi";
-  case def:
-    return "us";
-  default:
-    return "something bad happened, undefined sort type";
+namespace FPHelpers{
+    auto
+  swap_float_int(unsigned int val) -> float{
```

Comment:
Did you use 4.9?

I may have been confused from trying to do a template.

Good advice!

On Wednesday, August 24, 2016, Michael Bentley <notifications@github.com>
wrote:

> In perpVects/QFPHelpers.cpp
> <https://github.com/Geof23/QFP/pull/13#discussion_r76157862>:
>
> > -std::string
> > -getSortName(sort_t val){
> > -  switch(val){
> > -  case lt:
> > -    return "lt";
> > -  case gt:
> > -    return "gt";
> > -  case bi:
> > -    return "bi";
> > -  case def:
> > -    return "us";
> > -  default:
> > -    return "something bad happened, undefined sort type";
> > +namespace FPHelpers{
> > +    auto
> > +  swap_float_int(unsigned int val) -> float{
>
> For the auto:
> I tried the change, and no warning. It's common to have different return
> types as long as the arguments are different types.
>
> In this context, auto does NOT actually mean type deduction. Instead it
> allows you to specify a "trailing return type". The only reason that is
> ever useful is in using decltype() with some expression dependent on the
> arguments passed in, which is only useful for templates. There's a section
> on it starting on page 25 of my "Effective Modern C++" book if you'd like
> to take a look next time you're in.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Geof23/QFP/pull/13/files/57db3da0327a58ab91d6a9ee1c70501608c4db5b#r76157862>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFiXD8JfSK6GPIq_Gajz2wCO7lHiKneVks5qjNLjgaJpZM4Jr1FF>
> .
>